### PR TITLE
CTCP-2949-LRN MessageSender conflict error changes

### PR DESCRIPTION
**Pull request by akila-sankar**

### CODE REVIEW

File: app/v2/services/PersistenceService.scala [hunk 1]
```diff
@@ -115,6 +116,7 @@ class PersistenceServiceImpl @Inject() (persistenceConnector: PersistenceConnect
           movementResponse => Right(movementResponse)
         }
         .recover {
+          case UpstreamErrorResponse(_, CONFLICT, _, _) => Left(PersistenceError.ConflictError("(\"CC015C\" :: \"TransitOperation\" :: \"LRN\" :: Nil)"))
```
[the remark this answers is not in the document]
Comment (akila-sankar) [reply app/v2/services/PersistenceService.scala]: @drnaylor-oc I updated the error type as mentioned by you and message as well. Regarding the $lrn value, I updated the upstream error message to be returned as "LRN $lrn has previously been used and cannot be reused" but the result comes in a JSON string with message & code and needs to be parsed it here. shall I parse it here or please any other effective way of handling this

[the remark this answers is not in the document]
Comment (drnaylor-oc) [reply app/v2/services/PersistenceService.scala]: You could make the error from upstream look like this:

```json
{
    "code": "CONFLICT",
    "lrn": "lrn123",
    "message": "message here" 
}
```

and then yes, you can parse the Json message here and just pull out `__ \ "lrn"` -- saves string parsing.

[the remark this answers is not in the document]
Comment (akila-sankar) [reply app/v2/services/PersistenceService.scala]: Sure, will try this way

[the remark this answers is not in the document]
Comment (akila-sankar) [reply app/v2/services/PersistenceService.scala]: Yes updated

**Review by drnaylor-oc — APPROVED** [review]

**Pull request merged**